cli: Example_nodelocal failed

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1877066&tab=buildLog) on [master@8c928666de3f90cf17d81d70a3ee759c39e2ce27](https://github.com/cockroachdb/cockroach/commits/8c928666de3f90cf17d81d70a3ee759c39e2ce27):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:33711 --insecure=false --certs-dir=/tmp/cli-test933351062 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:33711 --insecure=false --certs-dir=/tmp/cli-test933351062 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:33711 --insecure=false --certs-dir=/tmp/cli-test933351062 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200415 05:31:09.561508 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:33711 --insecure=false --certs-dir=/tmp/cli-test933351062 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200415 05:31:09.563182 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:33711 --insecure=false --certs-dir=/tmp/cli-test933351062 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200415 05:31:09.564796 1 cli/error.go:373  open notexist.csv: no such file or directory
W200415 05:31:09.565069 2137419 kv/kvserver/intentresolver/intent_resolver.go:745  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.15s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1881799&tab=buildLog) on [master@a63a670b1a591be2ed38af7f98c1f85e32546596](https://github.com/cockroachdb/cockroach/commits/a63a670b1a591be2ed38af7f98c1f85e32546596):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:38041 --insecure=false --certs-dir=/tmp/cli-test229950652 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:38041 --insecure=false --certs-dir=/tmp/cli-test229950652 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:38041 --insecure=false --certs-dir=/tmp/cli-test229950652 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200417 05:34:03.032502 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:38041 --insecure=false --certs-dir=/tmp/cli-test229950652 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200417 05:34:03.034280 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:38041 --insecure=false --certs-dir=/tmp/cli-test229950652 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200417 05:34:03.035568 1 cli/error.go:373  open notexist.csv: no such file or directory
W200417 05:34:03.035757 1252867 kv/kvserver/intentresolver/intent_resolver.go:745  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.16s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1884827&tab=buildLog) on [master@ce5d511368b741555344ea28efa40fb9facfa577](https://github.com/cockroachdb/cockroach/commits/ce5d511368b741555344ea28efa40fb9facfa577):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:42221 --insecure=false --certs-dir=/tmp/cli-test598896381 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:42221 --insecure=false --certs-dir=/tmp/cli-test598896381 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200419 05:54:41.382816 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:42221 --insecure=false --certs-dir=/tmp/cli-test598896381 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200419 05:54:41.384534 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:42221 --insecure=false --certs-dir=/tmp/cli-test598896381 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200419 05:54:41.386074 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:42221 --insecure=false --certs-dir=/tmp/cli-test598896381 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200419 05:54:41.387537 1 cli/error.go:373  open notexist.csv: no such file or directory
W200419 05:54:41.387717 843757 kv/kvserver/intentresolver/intent_resolver.go:745  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.20s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1884828&tab=buildLog) on [master@ce5d511368b741555344ea28efa40fb9facfa577](https://github.com/cockroachdb/cockroach/commits/ce5d511368b741555344ea28efa40fb9facfa577):

```
=== RUN   Example_nodelocal
W200419 06:16:18.006440 1 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W200419 06:16:18.127029 263539 kv/kvserver/store.go:1627  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
[nodelocal --host=127.0.0.1:45471 --insecure=false --certs-dir=/tmp/cli-test188685309 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:45471 --insecure=false --certs-dir=/tmp/cli-test188685309 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:45471 --insecure=false --certs-dir=/tmp/cli-test188685309 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200419 06:16:19.048119 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:45471 --insecure=false --certs-dir=/tmp/cli-test188685309 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200419 06:16:19.055475 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45471 --insecure=false --certs-dir=/tmp/cli-test188685309 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200419 06:16:19.064864 1 cli/error.go:373  open notexist.csv: no such file or directory
W200419 06:16:19.066154 263885 kv/kvserver/intentresolver/intent_resolver.go:745  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (1.11s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1885970&tab=buildLog) on [master@666a0ac62832b6884bc6b039b4c944dbb42924aa](https://github.com/cockroachdb/cockroach/commits/666a0ac62832b6884bc6b039b4c944dbb42924aa):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:35105 --insecure=false --certs-dir=/tmp/cli-test448703468 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:35105 --insecure=false --certs-dir=/tmp/cli-test448703468 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:35105 --insecure=false --certs-dir=/tmp/cli-test448703468 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200420 05:31:58.358432 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:35105 --insecure=false --certs-dir=/tmp/cli-test448703468 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200420 05:31:58.359800 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:35105 --insecure=false --certs-dir=/tmp/cli-test448703468 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200420 05:31:58.361047 1 cli/error.go:373  open notexist.csv: no such file or directory
W200420 05:31:58.361169 2277213 kv/kvserver/intentresolver/intent_resolver.go:745  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.18s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1900898&tab=buildLog) on [master@60c9e055e970bd7f150ebcfad266929b2638d635](https://github.com/cockroachdb/cockroach/commits/60c9e055e970bd7f150ebcfad266929b2638d635):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:37827 --insecure=false --certs-dir=/tmp/cli-test193934596 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:37827 --insecure=false --certs-dir=/tmp/cli-test193934596 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:37827 --insecure=false --certs-dir=/tmp/cli-test193934596 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200426 05:40:43.638620 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:37827 --insecure=false --certs-dir=/tmp/cli-test193934596 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200426 05:40:43.639861 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:37827 --insecure=false --certs-dir=/tmp/cli-test193934596 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200426 05:40:43.641046 1 cli/error.go:373  open notexist.csv: no such file or directory
W200426 05:40:43.641187 2088184 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.14s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1902263&tab=buildLog) on [master@a20a8811ee6abfe3754220e893ed383afbab21c9](https://github.com/cockroachdb/cockroach/commits/a20a8811ee6abfe3754220e893ed383afbab21c9):

```
=== RUN   Example_nodelocal
W200427 05:51:12.249477 1 server/status/runtime.go:308  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W200427 05:51:12.389856 242125 kv/kvserver/store.go:1627  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
[nodelocal --host=127.0.0.1:43515 --insecure=false --certs-dir=/tmp/cli-test727893372 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:43515 --insecure=false --certs-dir=/tmp/cli-test727893372 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:43515 --insecure=false --certs-dir=/tmp/cli-test727893372 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200427 05:51:13.301267 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:43515 --insecure=false --certs-dir=/tmp/cli-test727893372 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200427 05:51:13.307463 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:43515 --insecure=false --certs-dir=/tmp/cli-test727893372 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200427 05:51:13.316177 1 cli/error.go:373  open notexist.csv: no such file or directory
W200427 05:51:13.317012 242683 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (1.10s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1904634&tab=buildLog) on [master@c3bb2bce3d69d0ceef20797d856a6ac107a3ef47](https://github.com/cockroachdb/cockroach/commits/c3bb2bce3d69d0ceef20797d856a6ac107a3ef47):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:45939 --insecure=false --certs-dir=/tmp/cli-test479428474 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:45939 --insecure=false --certs-dir=/tmp/cli-test479428474 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:45939 --insecure=false --certs-dir=/tmp/cli-test479428474 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200428 05:54:11.224562 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45939 --insecure=false --certs-dir=/tmp/cli-test479428474 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200428 05:54:11.226493 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45939 --insecure=false --certs-dir=/tmp/cli-test479428474 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200428 05:54:11.228005 1 cli/error.go:373  open notexist.csv: no such file or directory
W200428 05:54:11.228205 1318917 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.17s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1916626&tab=buildLog) on [master@9410f923dc346ba3c87fa129c39c8fdf59e601d0](https://github.com/cockroachdb/cockroach/commits/9410f923dc346ba3c87fa129c39c8fdf59e601d0):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:42549 --insecure=false --certs-dir=/tmp/cli-test399468657 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:42549 --insecure=false --certs-dir=/tmp/cli-test399468657 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200503 06:44:33.149628 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:42549 --insecure=false --certs-dir=/tmp/cli-test399468657 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200503 06:44:33.151274 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:42549 --insecure=false --certs-dir=/tmp/cli-test399468657 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200503 06:44:33.152701 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:42549 --insecure=false --certs-dir=/tmp/cli-test399468657 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200503 06:44:33.154305 1 cli/error.go:373  open notexist.csv: no such file or directory
W200503 06:44:33.154478 908978 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.15s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1918274&tab=buildLog) on [master@96f8f33dceeec0ad6168e2dffa98b952892bdf7b](https://github.com/cockroachdb/cockroach/commits/96f8f33dceeec0ad6168e2dffa98b952892bdf7b):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:39735 --insecure=false --certs-dir=/tmp/cli-test025502389 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:39735 --insecure=false --certs-dir=/tmp/cli-test025502389 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:39735 --insecure=false --certs-dir=/tmp/cli-test025502389 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200504 05:33:33.548774 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:39735 --insecure=false --certs-dir=/tmp/cli-test025502389 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200504 05:33:33.550343 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:39735 --insecure=false --certs-dir=/tmp/cli-test025502389 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200504 05:33:33.552104 1 cli/error.go:373  open notexist.csv: no such file or directory
W200504 05:33:33.552468 2107214 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.19s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1920823&tab=buildLog) on [master@425eaa8fb05fc32b2c42827b85338daa52f4177c](https://github.com/cockroachdb/cockroach/commits/425eaa8fb05fc32b2c42827b85338daa52f4177c):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:46083 --insecure=false --certs-dir=/tmp/cli-test215907280 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:46083 --insecure=false --certs-dir=/tmp/cli-test215907280 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:46083 --insecure=false --certs-dir=/tmp/cli-test215907280 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200505 05:49:24.154276 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:46083 --insecure=false --certs-dir=/tmp/cli-test215907280 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200505 05:49:24.156077 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:46083 --insecure=false --certs-dir=/tmp/cli-test215907280 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200505 05:49:24.161257 1 cli/error.go:373  open notexist.csv: no such file or directory
W200505 05:49:24.161413 665080 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.23s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1928595&tab=buildLog) on [master@f89c93fa20887f8d269149d8bc573ba8e00c99e2](https://github.com/cockroachdb/cockroach/commits/f89c93fa20887f8d269149d8bc573ba8e00c99e2):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:34159 --insecure=false --certs-dir=/tmp/cli-test220348602 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:34159 --insecure=false --certs-dir=/tmp/cli-test220348602 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:34159 --insecure=false --certs-dir=/tmp/cli-test220348602 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200508 05:54:07.319091 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34159 --insecure=false --certs-dir=/tmp/cli-test220348602 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200508 05:54:07.320701 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34159 --insecure=false --certs-dir=/tmp/cli-test220348602 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200508 05:54:07.322179 1 cli/error.go:373  open notexist.csv: no such file or directory
W200508 05:54:07.322407 899971 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.17s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1932499&tab=buildLog) on [master@c1abb272c94a437f1df9175fc30dc6a6729d3338](https://github.com/cockroachdb/cockroach/commits/c1abb272c94a437f1df9175fc30dc6a6729d3338):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:45903 --insecure=false --certs-dir=/tmp/cli-test587608800 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200510 05:34:12.356125 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45903 --insecure=false --certs-dir=/tmp/cli-test587608800 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200510 05:34:12.357649 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45903 --insecure=false --certs-dir=/tmp/cli-test587608800 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200510 05:34:12.358930 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45903 --insecure=false --certs-dir=/tmp/cli-test587608800 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200510 05:34:12.360227 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45903 --insecure=false --certs-dir=/tmp/cli-test587608800 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200510 05:34:12.361410 1 cli/error.go:373  open notexist.csv: no such file or directory
W200510 05:34:12.361566 750804 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.09s)
got:
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1934150&tab=buildLog) on [master@e6b47088aba9c9978501b966a0e88aeb273d9990](https://github.com/cockroachdb/cockroach/commits/e6b47088aba9c9978501b966a0e88aeb273d9990):

```
=== RUN   Example_nodelocal
W200511 06:31:59.824654 1 server/status/runtime.go:309  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W200511 06:31:59.955047 280602 kv/kvserver/store.go:1627  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
W200511 06:32:00.012579 280602 kv/kvserver/store.go:1627  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
[nodelocal --host=127.0.0.1:33297 --insecure=false --certs-dir=/tmp/cli-test744800705 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:33297 --insecure=false --certs-dir=/tmp/cli-test744800705 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200511 06:32:00.769999 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:33297 --insecure=false --certs-dir=/tmp/cli-test744800705 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200511 06:32:00.775967 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:33297 --insecure=false --certs-dir=/tmp/cli-test744800705 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200511 06:32:00.780807 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:33297 --insecure=false --certs-dir=/tmp/cli-test744800705 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200511 06:32:00.785603 1 cli/error.go:373  open notexist.csv: no such file or directory
W200511 06:32:00.786336 281140 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (1.01s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1938892&tab=buildLog) on [master@3531318daf434670bb23ae68b8e8b8a4e4801879](https://github.com/cockroachdb/cockroach/commits/3531318daf434670bb23ae68b8e8b8a4e4801879):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:39219 --insecure=false --certs-dir=/tmp/cli-test634105206 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:39219 --insecure=false --certs-dir=/tmp/cli-test634105206 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:39219 --insecure=false --certs-dir=/tmp/cli-test634105206 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200513 06:37:10.298826 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:39219 --insecure=false --certs-dir=/tmp/cli-test634105206 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200513 06:37:10.300576 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:39219 --insecure=false --certs-dir=/tmp/cli-test634105206 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200513 06:37:10.302300 1 cli/error.go:373  open notexist.csv: no such file or directory
W200513 06:37:10.302535 779786 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W200513 06:37:10.302543 779935 kv/txn.go:607  [n1,client=127.0.0.1:50872,hostssl,user=root] failure aborting transaction: node unavailable; try another peer; abort caused by: connExecutor closing
--- FAIL: Example_nodelocal (0.19s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1942100&tab=buildLog) on [master@752c32580ff452bb0039e5af0c00044b91bdcb12](https://github.com/cockroachdb/cockroach/commits/752c32580ff452bb0039e5af0c00044b91bdcb12):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:40047 --insecure=false --certs-dir=/tmp/cli-test557208235 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:40047 --insecure=false --certs-dir=/tmp/cli-test557208235 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:40047 --insecure=false --certs-dir=/tmp/cli-test557208235 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200514 05:54:16.846624 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:40047 --insecure=false --certs-dir=/tmp/cli-test557208235 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200514 05:54:16.847970 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:40047 --insecure=false --certs-dir=/tmp/cli-test557208235 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200514 05:54:16.849214 1 cli/error.go:373  open notexist.csv: no such file or directory
W200514 05:54:16.849438 726929 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.17s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1944892&tab=buildLog) on [master@68ca7367a1bf4c986e7f85de71c099b46bc435ab](https://github.com/cockroachdb/cockroach/commits/68ca7367a1bf4c986e7f85de71c099b46bc435ab):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:39885 --insecure=false --certs-dir=/tmp/cli-test867213248 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:39885 --insecure=false --certs-dir=/tmp/cli-test867213248 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200515 11:16:09.806926 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:39885 --insecure=false --certs-dir=/tmp/cli-test867213248 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200515 11:16:09.808637 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:39885 --insecure=false --certs-dir=/tmp/cli-test867213248 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200515 11:16:09.810073 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:39885 --insecure=false --certs-dir=/tmp/cli-test867213248 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200515 11:16:09.811361 1 cli/error.go:373  open notexist.csv: no such file or directory
W200515 11:16:09.811556 767950 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.11s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1947479&tab=buildLog) on [master@5156843cc23adecb6a70fabf19f51f46de1241ec](https://github.com/cockroachdb/cockroach/commits/5156843cc23adecb6a70fabf19f51f46de1241ec):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:35151 --insecure=false --certs-dir=/tmp/cli-test997931747 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:35151 --insecure=false --certs-dir=/tmp/cli-test997931747 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:35151 --insecure=false --certs-dir=/tmp/cli-test997931747 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200516 05:24:30.620669 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:35151 --insecure=false --certs-dir=/tmp/cli-test997931747 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200516 05:24:30.623917 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:35151 --insecure=false --certs-dir=/tmp/cli-test997931747 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200516 05:24:30.625349 1 cli/error.go:373  open notexist.csv: no such file or directory
W200516 05:24:30.625500 801204 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W200516 05:24:30.625609 800461 ts/db.go:194  [n1,ts-poll] error writing time series data: failed to send RPC: sending to all 1 replicas failed; last error: (err: [NotLeaseHolderError] r4: replica (n1,s1):1 not lease holder; lease holder unknown) <nil>
--- FAIL: Example_nodelocal (0.18s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1955712&tab=buildLog) on [master@8fec3f4c6d136a86f472c975edd36b75e5ab9a8c](https://github.com/cockroachdb/cockroach/commits/8fec3f4c6d136a86f472c975edd36b75e5ab9a8c):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:41481 --insecure=false --certs-dir=/tmp/cli-test663456497 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:41481 --insecure=false --certs-dir=/tmp/cli-test663456497 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:41481 --insecure=false --certs-dir=/tmp/cli-test663456497 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200520 05:35:41.514838 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:41481 --insecure=false --certs-dir=/tmp/cli-test663456497 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200520 05:35:41.516595 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:41481 --insecure=false --certs-dir=/tmp/cli-test663456497 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200520 05:35:41.520359 1 cli/error.go:373  open notexist.csv: no such file or directory
W200520 05:35:41.520596 733178 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.22s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1957931&tab=buildLog) on [master@1520ad2ba7c926f8043de8b6e044ab35c2f67b13](https://github.com/cockroachdb/cockroach/commits/1520ad2ba7c926f8043de8b6e044ab35c2f67b13):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:38231 --insecure=false --certs-dir=/tmp/cli-test158871281 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:38231 --insecure=false --certs-dir=/tmp/cli-test158871281 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:38231 --insecure=false --certs-dir=/tmp/cli-test158871281 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200521 06:15:26.869039 1 cli/error.go:373  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:38231 --insecure=false --certs-dir=/tmp/cli-test158871281 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200521 06:15:26.870340 1 cli/error.go:373  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:38231 --insecure=false --certs-dir=/tmp/cli-test158871281 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200521 06:15:26.871529 1 cli/error.go:373  open notexist.csv: no such file or directory
W200521 06:15:26.871681 724568 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.11s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1960727&tab=buildLog) on [master@d7985c53f1e6d32abd604f5987b9a44f6b210ee3](https://github.com/cockroachdb/cockroach/commits/d7985c53f1e6d32abd604f5987b9a44f6b210ee3):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:35657 --insecure=false --certs-dir=/tmp/cli-test817464391 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:35657 --insecure=false --certs-dir=/tmp/cli-test817464391 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200522 05:32:07.552500 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:35657 --insecure=false --certs-dir=/tmp/cli-test817464391 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200522 05:32:07.553965 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:35657 --insecure=false --certs-dir=/tmp/cli-test817464391 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200522 05:32:07.555385 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:35657 --insecure=false --certs-dir=/tmp/cli-test817464391 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200522 05:32:07.557282 1 cli/error.go:374  open notexist.csv: no such file or directory
W200522 05:32:07.557455 715675 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.14s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1962692&tab=buildLog) on [master@d5dd67bc846b73cebe8dc73b320de045ce8a170c](https://github.com/cockroachdb/cockroach/commits/d5dd67bc846b73cebe8dc73b320de045ce8a170c):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:33867 --insecure=false --certs-dir=/tmp/cli-test556531229 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:33867 --insecure=false --certs-dir=/tmp/cli-test556531229 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:33867 --insecure=false --certs-dir=/tmp/cli-test556531229 upload test.csv /test/file1.csv]
*
* ERROR: pq: destination file already exists for /test/file1.csv
*
E200523 05:40:32.965040 1 cli/error.go:374  pq: destination file already exists for /test/file1.csv
[nodelocal --host=127.0.0.1:33867 --insecure=false --certs-dir=/tmp/cli-test556531229 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200523 05:40:32.966739 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:33867 --insecure=false --certs-dir=/tmp/cli-test556531229 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200523 05:40:32.967944 1 cli/error.go:374  open notexist.csv: no such file or directory
W200523 05:40:32.968067 747252 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.16s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1962693&tab=buildLog) on [master@d5dd67bc846b73cebe8dc73b320de045ce8a170c](https://github.com/cockroachdb/cockroach/commits/d5dd67bc846b73cebe8dc73b320de045ce8a170c):

```
=== RUN   Example_nodelocal
W200523 05:45:52.499886 1 server/status/runtime.go:309  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W200523 05:45:52.616556 276278 kv/kvserver/store.go:1652  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
[nodelocal --host=127.0.0.1:45529 --insecure=false --certs-dir=/tmp/cli-test196728994 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:45529 --insecure=false --certs-dir=/tmp/cli-test196728994 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200523 05:45:53.421958 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45529 --insecure=false --certs-dir=/tmp/cli-test196728994 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200523 05:45:53.428937 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45529 --insecure=false --certs-dir=/tmp/cli-test196728994 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200523 05:45:53.433972 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45529 --insecure=false --certs-dir=/tmp/cli-test196728994 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200523 05:45:53.438889 1 cli/error.go:374  open notexist.csv: no such file or directory
W200523 05:45:53.439698 277013 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.96s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1967719&tab=buildLog) on [master@8f7cd615bc0cf910816d81905c0e683f9748c77d](https://github.com/cockroachdb/cockroach/commits/8f7cd615bc0cf910816d81905c0e683f9748c77d):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:44533 --insecure=false --certs-dir=/tmp/cli-test018938405 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:44533 --insecure=false --certs-dir=/tmp/cli-test018938405 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:44533 --insecure=false --certs-dir=/tmp/cli-test018938405 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200526 05:36:00.801095 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:44533 --insecure=false --certs-dir=/tmp/cli-test018938405 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200526 05:36:00.802146 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:44533 --insecure=false --certs-dir=/tmp/cli-test018938405 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200526 05:36:00.803264 1 cli/error.go:374  open notexist.csv: no such file or directory
W200526 05:36:00.803429 650846 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.09s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1976809&tab=buildLog) on [master@f6ce99f3e8c70059afdcb7683432cde5007c5102](https://github.com/cockroachdb/cockroach/commits/f6ce99f3e8c70059afdcb7683432cde5007c5102):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:37365 --insecure=false --certs-dir=/tmp/cli-test403444134 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:37365 --insecure=false --certs-dir=/tmp/cli-test403444134 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200529 06:14:10.222139 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:37365 --insecure=false --certs-dir=/tmp/cli-test403444134 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200529 06:14:10.223441 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:37365 --insecure=false --certs-dir=/tmp/cli-test403444134 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200529 06:14:10.224714 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:37365 --insecure=false --certs-dir=/tmp/cli-test403444134 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200529 06:14:10.225995 1 cli/error.go:374  open notexist.csv: no such file or directory
W200529 06:14:10.226165 757761 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.11s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1983968&tab=buildLog) on [master@d65b224bc7c8b99effd42f14677aa066ddc94cdb](https://github.com/cockroachdb/cockroach/commits/d65b224bc7c8b99effd42f14677aa066ddc94cdb):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:34357 --insecure=false --certs-dir=/tmp/cli-test277726389 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:34357 --insecure=false --certs-dir=/tmp/cli-test277726389 upload test.csv /test/file2.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200602 05:37:19.615127 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34357 --insecure=false --certs-dir=/tmp/cli-test277726389 upload test.csv /test/file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200602 05:37:19.616616 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34357 --insecure=false --certs-dir=/tmp/cli-test277726389 upload test.csv /test/../../file1.csv]
*
* ERROR: open test.csv: no such file or directory
*
E200602 05:37:19.617736 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34357 --insecure=false --certs-dir=/tmp/cli-test277726389 upload notexist.csv /test/file1.csv]
*
* ERROR: open notexist.csv: no such file or directory
*
E200602 05:37:19.618917 1 cli/error.go:374  open notexist.csv: no such file or directory
W200602 05:37:19.619095 761514 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.17s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1992612&tab=buildLog) on [master@dfb6e1bee4abc2f2ff82d545686908d983f82872](https://github.com/cockroachdb/cockroach/commits/dfb6e1bee4abc2f2ff82d545686908d983f82872):

```
=== RUN   Example_nodelocal
W200606 05:45:35.123698 1 server/status/runtime.go:309  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W200606 05:45:35.245142 291249 kv/kvserver/store.go:1652  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
[nodelocal --host=127.0.0.1:32873 --insecure=false --certs-dir=/tmp/cli-test619289507 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:32873 --insecure=false --certs-dir=/tmp/cli-test619289507 upload test.csv /test/file2.csv]
E200606 05:45:36.156747 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:32873 --insecure=false --certs-dir=/tmp/cli-test619289507 upload test.csv /test/file1.csv]
E200606 05:45:36.161925 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:32873 --insecure=false --certs-dir=/tmp/cli-test619289507 upload test.csv /test/../../file1.csv]
E200606 05:45:36.167110 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:32873 --insecure=false --certs-dir=/tmp/cli-test619289507 upload notexist.csv /test/file1.csv]
E200606 05:45:36.172858 1 cli/error.go:374  open notexist.csv: no such file or directory
W200606 05:45:36.173819 292033 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (1.08s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=2

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1992611&tab=buildLog) on [master@dfb6e1bee4abc2f2ff82d545686908d983f82872](https://github.com/cockroachdb/cockroach/commits/dfb6e1bee4abc2f2ff82d545686908d983f82872):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:46049 --insecure=false --certs-dir=/tmp/cli-test635320624 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:46049 --insecure=false --certs-dir=/tmp/cli-test635320624 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:46049 --insecure=false --certs-dir=/tmp/cli-test635320624 upload test.csv /test/file1.csv]
E200606 06:11:43.984578 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:46049 --insecure=false --certs-dir=/tmp/cli-test635320624 upload test.csv /test/../../file1.csv]
E200606 06:11:43.986077 1 cli/error.go:374  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:46049 --insecure=false --certs-dir=/tmp/cli-test635320624 upload notexist.csv /test/file1.csv]
E200606 06:11:43.987575 1 cli/error.go:374  open notexist.csv: no such file or directory
W200606 06:11:43.987793 846332 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.14s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1994140&tab=buildLog) on [master@3bf61fdac388e3807edd15feaf44028c0398c2ec](https://github.com/cockroachdb/cockroach/commits/3bf61fdac388e3807edd15feaf44028c0398c2ec):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:34459 --insecure=false --certs-dir=/tmp/cli-test534413231 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:34459 --insecure=false --certs-dir=/tmp/cli-test534413231 upload test.csv /test/file2.csv]
E200607 05:44:42.736311 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34459 --insecure=false --certs-dir=/tmp/cli-test534413231 upload test.csv /test/file1.csv]
E200607 05:44:42.738705 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34459 --insecure=false --certs-dir=/tmp/cli-test534413231 upload test.csv /test/../../file1.csv]
E200607 05:44:42.740039 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:34459 --insecure=false --certs-dir=/tmp/cli-test534413231 upload notexist.csv /test/file1.csv]
E200607 05:44:42.741434 1 cli/error.go:375  open notexist.csv: no such file or directory
W200607 05:44:42.741670 859070 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.10s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2001971&tab=buildLog) on [master@c5e5f5f88a6b5dfb89585e53a0a39414b225717c](https://github.com/cockroachdb/cockroach/commits/c5e5f5f88a6b5dfb89585e53a0a39414b225717c):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:45071 --insecure=false --certs-dir=/tmp/cli-test017915143 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:45071 --insecure=false --certs-dir=/tmp/cli-test017915143 upload test.csv /test/file2.csv]
E200610 05:45:00.629105 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45071 --insecure=false --certs-dir=/tmp/cli-test017915143 upload test.csv /test/file1.csv]
E200610 05:45:00.630485 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45071 --insecure=false --certs-dir=/tmp/cli-test017915143 upload test.csv /test/../../file1.csv]
E200610 05:45:00.631684 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:45071 --insecure=false --certs-dir=/tmp/cli-test017915143 upload notexist.csv /test/file1.csv]
E200610 05:45:00.632850 1 cli/error.go:375  open notexist.csv: no such file or directory
W200610 05:45:00.633023 809826 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.16s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).Example_nodelocal failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2004862&tab=buildLog) on [master@49339fef48f4165a4e7471290bb08d422cf835a9](https://github.com/cockroachdb/cockroach/commits/49339fef48f4165a4e7471290bb08d422cf835a9):

```
=== RUN   Example_nodelocal
[nodelocal --host=127.0.0.1:40443 --insecure=false --certs-dir=/tmp/cli-test734148700 upload test.csv /test/file1.csv]
[nodelocal --host=127.0.0.1:40443 --insecure=false --certs-dir=/tmp/cli-test734148700 upload test.csv /test/file2.csv]
[nodelocal --host=127.0.0.1:40443 --insecure=false --certs-dir=/tmp/cli-test734148700 upload test.csv /test/file1.csv]
E200611 05:40:15.967908 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:40443 --insecure=false --certs-dir=/tmp/cli-test734148700 upload test.csv /test/../../file1.csv]
E200611 05:40:15.969423 1 cli/error.go:375  open test.csv: no such file or directory
[nodelocal --host=127.0.0.1:40443 --insecure=false --certs-dir=/tmp/cli-test734148700 upload notexist.csv /test/file1.csv]
E200611 05:40:15.970632 1 cli/error.go:375  open notexist.csv: no such file or directory
W200611 05:40:15.970845 770234 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
--- FAIL: Example_nodelocal (0.13s)
got:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload test.csv /test/../../file1.csv
ERROR: open test.csv: no such file or directory
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory
want:
nodelocal upload test.csv /test/file1.csv
successfully uploaded to nodelocal://1/test/file1.csv
nodelocal upload test.csv /test/file2.csv
successfully uploaded to nodelocal://1/test/file2.csv
nodelocal upload test.csv /test/file1.csv
ERROR: destination file already exists for /test/file1.csv
nodelocal upload test.csv /test/../../file1.csv
ERROR: local file access to paths outside of external-io-dir is not allowed: /test/../../file1.csv
nodelocal upload notexist.csv /test/file1.csv
ERROR: open notexist.csv: no such file or directory

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=Example_nodelocal PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2AExample_nodelocal.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/cli/cli_test.go
+++ b/pkg/cli/cli_test.go
@@ -70,6 +70,11 @@ type cliTestParams struct {
 	noNodelocal bool
 }
 
+// testTempFilePrefix is a sentinel marker to be used as the prefix of a
+// test file name. It is used to extract the file name from a uniquely
+// generated (temp directory) file path.
+const testTempFilePrefix = "test-temp-prefix-"
+
 func (c *cliTest) fail(err interface{}) {
 	if c.t != nil {
 		defer c.logScope.Close(c.t)
@@ -332,6 +337,16 @@ func (c cliTest) RunWithArgs(origArgs []string) {
 			}
 		}
 		args = append(args, origArgs[1:]...)
+
+		// `nodelocal upload` CLI tests create test files in unique temp
+		// directories. Given that the expected output for such tests is defined as
+		// a static comment, it is not possible to match against the full file path.
+		// So, we trim the file path upto the sentinel prefix marker, and use only
+		// the file name for comparing against the expected output.
+		if len(origArgs) >= 3 && strings.Contains(origArgs[2], testTempFilePrefix) {
+			splitFilePath := strings.Split(origArgs[2], testTempFilePrefix)
+			origArgs[2] = splitFilePath[1]
+		}
 
 		if !c.omitArgs {
 			fmt.Fprintf(os.Stderr, "%s\n", args)

--- a/pkg/cli/nodelocal_test.go
+++ b/pkg/cli/nodelocal_test.go
@@ -126,11 +126,15 @@ func TestNodeLocalFileUpload(t *testing.T) {
 }
 
 func createTestFile(name, content string) (string, func()) {
-	err := ioutil.WriteFile(name, []byte(content), 0666)
+	tmpDir, err := ioutil.TempDir("", "")
+	tmpFile := filepath.Join(tmpDir, testTempFilePrefix+name)
+	if err == nil {
+		err = ioutil.WriteFile(tmpFile, []byte(content), 0666)
+	}
 	if err != nil {
 		return "", func() {}
 	}
-	return name, func() {
-		_ = os.Remove(name)
+	return tmpFile, func() {
+		_ = os.RemoveAll(tmpDir)
 	}
 }

